evaluator/builtin_string.go

```diff
@@ -603,3 +603,49 @@ func trimRight(str, remstr string) string {
 		str = x
 	}
 }
+
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_rpad
+func builtinRpad(args []types.Datum, ctx context.Context) (d types.Datum, err error) {
+	// RPAD(str,len,padstr)
+	// args[0] string, args[1] int, args[2] string
+	str, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	length, err := args[1].ToInt64(ctx.GetSessionVars().StmtCtx)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	l := int(length)
+
+	padStr, err := args[2].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+
```

Comment:
IMHO, the following logic too much complex than it should be, and the loop is  hard to read.
Here is a simpler example:
```
func test(str string, length int, pad string) {
	for len(str) < length {
		str = str + pad
	}
	return str[:length]
}
```

just check `pad != ""` before this logic.